api/v1/openapi.json

```diff
@@ -0,0 +1,1115 @@
+			"get": {
+				"summary": "Retrieve an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{ "$ref": "#/parameters/endpoint-id" },
+					{ "$ref": "#/parameters/endpoint-id-type" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Endpoint"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"put": {
+				"summary": "Create or update an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{
+						"name": "endpoint",
+						"in": "body",
+						"required": true,
+						"schema": {
+							"$ref": "#/definitions/Endpoint"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "Success",
+						"schema": {
+							"$ref": "#/definitions/EndpointResponse"
+						}
+					},
+                                        "405": {
+                                                "description": "Endpoint already exists",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        },
+                                        "default": {
+                                                "description": "Unexpected error",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"delete": {
+				"description": "Delete an endpoint",
+				"tags": [ "endpoint" ],
+				"parameters": [
+					{ "$ref": "#/parameters/endpoint-id" },
+					{ "$ref": "#/parameters/endpoint-id-type" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Endpoint"
+                                                }
+					},
+                                        "404": {
+                                                "description": "Endpoint not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
```

Comment:
Deletion is imminent and atomic. Creation is not atomic because there is the state of creation the endopint, then retrieving of metadata followed by program generation and attachment until ready. Will document this.